src/responder/common/cache_req/plugins/cache_req_user_by_filter.c

```diff
@@ -86,10 +86,19 @@ cache_req_user_by_filter_lookup(TALLOC_CTX *mem_ctx,
     char *recent_filter;
     errno_t ret;
 
-    recent_filter = talloc_asprintf(mem_ctx, "(%s>=%lu)", SYSDB_LAST_UPDATE,
-                                    cr->req_start);
-    if (recent_filter == NULL) {
-        return ENOMEM;
+    /* The "files" provider updates the record if /etc/passwd or /etc/group
+     * is touched. It does not perform any per-request update.
+     * Therefore the last update flag is not updated if no file was touched
+     * and we cannot use this optimization.
+     */
+    if (strcasecmp(domain->provider, "files") == 0) {
```

Comment:
Would it be possible to use `is_files_provider()` from `util.h`?

Reply:
Nice. Didn't know that function existed. Thanks.